notesapi/v1/tests/test_views.py

```diff
@@ -274,121 +355,136 @@ def test_delete_notfound(self):
-        result = self._get_search_results('offset=foobar')
-        self.assertEqual(len(result['rows']), settings.RESULTS_DEFAULT_SIZE)
-        self.assertEqual(result['rows'][0], first)
+        results = self._get_search_results(text=u"Свят")
+        self.assertEqual(results['rows'][0]['text'], u'Веселих свят')
 
-    def test_read_all_no_annotations(self):
+    def test_search_multiword(self):
         """
-        Tests list all annotations endpoint when no annotations are present in elasticsearch.
+        Tests searching of complex words and word combinations
         """
-        url = reverse('api:v1:annotations')
-        response = self.client.get(url, self.headers)
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertEqual(len(response.data), 0, "no annotation should be returned in response")
+        self._create_annotation(text=u'Totally different something')
+        self.assertEqual(self._get_search_results(text=u"TOTALLY")['total'], 1)
+        self.assertEqual(self._get_search_results(text=u"different")['total'], 1)
+        self.assertEqual(self._get_search_results(text=u"differ")['total'], 1)
+        self.assertEqual(self._get_search_results(text=u"total")['total'], 1)
+        self.assertEqual(self._get_search_results(text=u"totil")['total'], 0)
+        self.assertEqual(self._get_search_results(text=u"something")['total'], 1)
+        self.assertEqual(self._get_search_results(text=u"totally different")['total'], 1)
 
-    def test_read_all(self):
+    def test_search_course(self):
         """
-        Tests list all annotations.
+        Tests searching with course_id provided
         """
-        for i in xrange(5):
-            kwargs = {'text': 'Foo_{}'.format(i), 'id': str(i)}
-            self._create_annotation(refresh=False, **kwargs)
+        self._create_annotation(text=u'First one', course_id="u'edX/DemoX/Demo_Course'")
+        self._create_annotation(text=u'Not shown', course_id="u'edx/demox/demo_course'")  # wrong case
+        self._create_annotation(text=u'Second note', course_id="u'edX/DemoX/Demo_Course'")
+        self._create_annotation(text=u'Third note', course_id="b")
+        self._create_annotation(text=u'Fourth note', course_id="c")
 
-        es.conn.indices.refresh(es.index)
+        results = self._get_search_results(course_id="u'edX/DemoX/Demo_Course'")
+        self.assertEqual(results['total'], 2)
 
-        url = reverse('api:v1:annotations')
-        response = self.client.get(url, self.headers)
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertEqual(len(response.data), 5, "five annotations should be returned in response")
+        results = self._get_search_results(course_id="b")
+        self.assertEqual(results['total'], 1)
+        self.assertEqual(results['rows'][0]['text'], u'Third note')
 
 
 @patch('django.conf.settings.DISABLE_TOKEN_CHECK', True)
 class AllowAllAnnotationViewTests(BaseAnnotationViewTests):
```

Comment:
no, for all that is wrong with tokens we send 403. As for this one - it is related to a particular note-saving issue, so I think it is more fit in API tests.